accepted/0000-package-overrides.md

```diff
@@ -0,0 +1,35 @@
+# Implement a package override option
+
+## Summary
+
+Add a section to package.json that allows the user to designate packages as an 'override' at a particular version.
+
+## Motivation
+
+In some situations, allowing more than one version of a package to be installed at the same time can cause unwanted behaviors. This will allow users to fix that by explicitly setting a single version to use.
+
+## Detailed Explanation
+
+Users will be able to add an additional dependency section called 'overrides' where they give a list of one or more packages to be installed at a specific version. Having this section will ensure that the only version of that package that is eventually installed to the project is the one set as an override. This will not affect the name, source, path, or other aspects of the package's installation.
+
+## Rationale and Alternatives
+
+Users don't frequently need to do this sort of override because npm's nested tree structure allows for a collection of dependencies to require incompatible semvers for the same package. In many cases, the default behavior is acceptable and nothing further is needed. In other, more problematic situations, the user may encounter conflicting binaries or other components that cannot be used simultaneously. The only current options are to attempt to get the maintainers of the dependencies to make changes that will reduce these conflicts, or to use a tool like `https://www.npmjs.com/package/replace-deep-dep` to repair the requirements.
+
+## Implementation
+
+In order to do this, the npm installer will need to read in an overrides section from `package.json`. This will have a simple format such as `overrides: { 'foo': '1.2.3' }`. These values will need to be used at the end of the tree-building process to select the final version. (Is there also a pruning element to this?)
```

Comment:
What if i want to override one package with another - like underscore for lodash, etc?

It would be ideal if the RHS of this could also be a full `npm install` argument - ie, package name/url + @ + version.

Reply:
You would use package aliases, which are coming soon, so: `"undescore": "npm:lodash"`

Reply:
Awesome! How soon?

Reply:
the same soon as overrides